torchvision/prototype/transforms/_utils.py

```diff
@@ -33,13 +33,15 @@ def _setup_float_or_seq(arg: Union[float, Sequence[float]], name: str, req_size:
 
 
 def _check_fill_arg(fill: Union[FillType, Dict[Type, FillType]]) -> None:
-    if isinstance(fill, dict):
+    if type(fill) == dict:
+        # Do exact type check to avoid accepting default dicts from the user. DefaultDict values can be verified only
+        # at runtime not at construction type.
```

Comment:
Here I'm checking for exact simple dict. The reason we can't accept defaultdicts after this change is because there is no great way to verify at constructor time their default value. We can only check their non-default values. This means that users need to specify explicitly all types rather than doing something like `fill=defaultdict(lambda: 0, {features.Mask: 255}))`. IMO though this idiom is convenient, it can lead to surprises. This will align the behaviour with `ConvertDtype`, `TransposeDimensions` etc where we expect the user to explicitly define the values.